daft/recordbatch/micropartition.py

```diff
@@ -180,10 +180,6 @@ def to_ipc_stream(self) -> bytes:
     # Compute methods (MicroPartition -> MicroPartition)
     ###
 
-    def cast_to_schema(self, schema: Schema) -> MicroPartition:
-        """Casts a MicroPartition into the provided schema."""
-        return MicroPartition._from_pymicropartition(self._micropartition.cast_to_schema(schema._schema))
```

Comment:
Are we concerned about remove these public APIs?

Reply:
I don't think so? But I could be wrong.

We do not document any of the micropartition/recordbatch methods, and we only expose these classes in certain sinks such as `DataFrame.into_partitions`, where users would generally just convert it into arrow or some other type. I also wouldn't want users to use `cast_to_schema`, and if they really wanted to, they could implement it on their own.